core_lib/src/structure/filemanager.cpp

```diff
@@ -547,10 +569,19 @@ QString FileManager::backupPreviousFile(const QString& fileName)
         return "";
 
     QFileInfo info(fileName);
-    QString sBackupFile = info.completeBaseName() + ".backup." + info.suffix();
+    int backupCount = 1;
+    QDir directory(info.absoluteDir());
+    QString backupString = "backup";
+    for (QFileInfo info : directory.entryInfoList(QDir::Filter::Files)) {
+        if (info.completeBaseName().contains(backupString)) {
```

Comment:
The word "backup" is commonly used in terms of file naming. The chances are that users could put "backup" in their filename and then the `backupCount` number would be wrong.

I would suggest to check with full filenames such as `sBackupFile` below to get a correct backup count.

Reply:
Ah.. good point, I forgot about that! will fix.

Reply:
Fixed